MESH-5405-Create prefixed SEs

Punakshi opened the pull request:
### Checklist
🚨 Please review this repository's [contribution guidelines](./CONTRIBUTING.md).

- [ ] I've read and agree to the project's contribution guidelines.
- [ ] I'm requesting to **pull a topic/feature/bugfix branch**.
- [ ] I checked that my code additions will pass code linting checks and unit tests.
- [ ] I updated unit and integration tests (if applicable).
- [ ] I'm ready to notify the team of this contribution.

### Description
What does this change do and why?
Handles the canary, bluegreen, deployment to rollout migration

[Link to related ISSUE]

Thank you!

Commit:
MESH-MESH-5405-IdentityConfig-Struct-Edit

Commit:
Unit test fix

Commit:
Logic to add SE for bluegreen, canary, dep to rollout migration

Commit:
Logic to add SE for bluegreen, canary, dep to rollout migration

Commit:
Unit tests for prefixed SE

Commit:
Logic for prefixed SE and migration

Commit:
Merge branch 'MESH-5405-Create-Additional-SE' into MESH-5405-Create-Additional-SE-UnitTests

Commit:
Unit tests prefixed SE and migration

Commit:
Unit tests fixes for prefixed SE and migration

Commit:
Remove redundant code

Commit:
Refactor code

Commit:
Adding unit test

Commit:
Simplifying the logic for service discovery in operator

Punakshi commented on admiral/pkg/clusters/configwriter.go:
```diff
@@ -116,56 +146,50 @@ func getServiceEntryEndpoints(
+	ep := endpoint.DeepCopy()
+	if clientCluster == serverCluster {
+		if strings.HasPrefix(host, canaryPrefix) || strings.HasPrefix(host, previewPrefix) {
+			ep.Address = services[testServiceKey][0].Name + common.Sep + identityConfigEnvironment.Namespace + common.GetLocalDomainSuffix()
```
Nil check

Punakshi replied:
Rename testServiceKey to `testSvc`

rtay1188 commented:
Looks good - only thing is that need some comments to explain the service map structure and expectations of state syncer. Meaning, we should only have default and test keys, should only have one service in the array value for the test key, etc. (in getServiceEntryEndpoints)

Commit:
Update service key handling and improve endpoint logic in configwriter

- Changed `testServiceKey` from "canary" to "testSvc"
- Enhanced error handling to check for empty endpoints
- Added detailed comments explaining service key usage scenarios
- Improved logic to handle services with `testServiceKey` in endpoint assignment

Commit:
Merge branch 'master' into MESH-5405-Create-Additional-SE-UnitTests

# Conflicts:
#	admiral/pkg/clusters/configwriter.go

Commit:
[MESH-5405] Update unit tests and sample data to use 'test' instead of 'canary'
